### PR TITLE
Amplicon sequence generation

### PR DESCRIPTION
**Pull request by ThijsMaas**
Hi! 
This is Thijs from ENPICOM. We talked some weeks ago about including support for amplicon sequencing in ISS. We have developed and tested this functionality and would like your opinion on the changes proposed here.

This PR includes the following:
- A new argument `--sequence_type` which adds the 'amplicon' option to the default 'metagenomics' 
  - This changes the `read_start` and `read_end` logic, for amplicon sequences the read pair is generated from the actual start and end of the sequence.
- Cleaned up some unused imports, variables, parameters, etc...
- Added new tests for amplicon sequencing mode

### CODE REVIEW

File: iss/modeller.py [hunk 1]
```diff
@@ -110,15 +106,14 @@ def raw_qualities_to_histogram(qualities):
     # quals = util.split_list([i for i in zip(*qualities)], n_parts=cpus)
     cdfs_list = []
     for q in qualities:
-        numpy_log_handler = np.seterrcall(util.nplog)
         with np.errstate(under='ignore', divide='call'):
             try:
                 kde = stats.gaussian_kde(q, bw_method=0.2 / np.std(q, ddof=1))
-            except np.linalg.linalg.LinAlgError as e:
```
Comment (HadrienG) [inline iss/modeller.py:117]: Why changing the exception type here?

**Review by HadrienG — CHANGES_REQUESTED** [review]
My apologies from the delay!

File: iss/app.py [hunk 2]
```diff
@@ -557,6 +559,13 @@ def main():
         help='Output file path and prefix (Required)',
         required=True
     )
+    parser_gen.add_argument(
+        '--sequence_type',
+        '-t',
+        choices=['metagenomics', 'amplicon'],
```
[the remark this answers is not in the document]
Comment (ThijsMaas) [reply iss/app.py:561]: Good call, I added the default

File: iss/modeller.py (hunk 1 above)
Comment (ThijsMaas) [reply iss/modeller.py]: I noticed a wrong commit was targeted to this branch, which I just reverted. That one included some more debugging things, which is why I changed the exception type here.

**Pull request closed without merging**